defi/src/api2/scripts/_test2.ts

```diff
@@ -0,0 +1,197 @@
+        // }
+        i++
+        if (i % 100 === 0) log(i, 'tvlData.length', tvlData.length, !!hourlyItem, protocol.name)
+      } catch (e) {
+        console.error(e)
+      }
+    });
+  console.timeEnd('getAllProtocolItems')
+
+
+  /* console.time('allTvlData')
+  let limit = 100000
+  let offset = 0
+  const allTvlData: any[] = []
+  let lastLength
+  do {
+    lastLength = allTvlData.length
+    const res = await TABLES.DAILY_TVL.findAll({ limit, offset, raw: true, attributes: ['id', 'data', 'timestamp'] })
+    allTvlData.push(...res)
+    offset += limit
+    log('allTvlData.length', allTvlData.length)
+  } while (allTvlData.length % limit === 0 && allTvlData.length !== lastLength)
+  console.timeEnd('allTvlData')
+ */
+
+  /*   console.time('allTvlDataMap')
+    const protocolMap: any = {}
+    protocols.forEach((protocol: any) => protocolMap[protocol.id] = protocol)
+    allTvlData.forEach(({ id, data, timestamp }: any) => {
+      data.SK = timestamp
+      const protocol = protocolMap[id]
+      if (protocol) {
+        if (!Array.isArray(protocol.tvlData)) protocol.tvlData = []
+        protocol.tvlData.push(data)
+      }
+    })
+    protocols.forEach((protocol: any) => {
+      if (Array.isArray(protocol.tvlData))
+        protocol.tvlData.sort((a: any, b: any) => a.SK - b.SK)
+      const hourlyItem = latestProtocolItemsMap[protocol.id]
+      if (hourlyItem) {
+        if (!protocol.tvlData.length || protocol.tvlData[protocol.tvlData.length - 1].SK !== hourlyItem.SK)
+          protocol.tvlData.push(hourlyItem)
+      }
+      if (!Array.isArray(protocol.tvlData)) protocol.tvlData = []
+  
+    })
+    console.timeEnd('allTvlDataMap') */
+
+
+  console.time('write data')
+  fs.writeFileSync('../../allTvlData.json', JSON.stringify(metadata))
+  console.timeEnd('write data')
+}
+
+async function main1() {
+  // console.time('initCache')
+  // await initCache()
+  // console.timeEnd('initCache')
+
+  const data = fs.readFileSync('../../allTvlData.json', 'utf8')
+  const metadata = JSON.parse(data)
+  const { protocols, protocolDataMap } = metadata
+  const protocolMap: any = {}
+  protocols.forEach((protocol: any) => protocolMap[protocol.id] = protocol)
+  // console.log('protocols.length', protocols.length)
+  // const protocolsWithData = protocols.filter((protocol: any) => protocolDataMap[protocol.id]?.tvlData?.length)
+  // console.log('protocolsWithData.length', protocolsWithData.length)
+  const protocolList = protocols.map((i: any) => { // clone and remove doublecounted field
+    i = {...i}
+    delete i.doublecounted
+    return i
+  })
+
+  const processProtocolsOptions: getHistoricalTvlForAllProtocolsOptionalOptions = {
+    usePGCache: true,
+    protocolList,
+    getLastTvl: (protocol: any) => protocolDataMap[protocol.id]?.lastHourlyRecord,
+    getAllTvlData: (protocol: any) => protocolDataMap[protocol.id]?.tvlData,
+    getModule: (protocol: any) => ({
+      doublecounted: protocolMap[protocol.id]?.doublecounted,
```

Comment:
in updateMetadata() .doublecounted is deleted, why keep the list on cache different from the list on db cache? feels like thats very likely to cause issues

Reply:
`doublecounted`prop is deleted from (in memory) cache/protocolList because, all the props there are returned as part of protocol response, maybe not an issue if this is also returned on protocols data in /protocols & /protocol/:name routes?

at the same time, some part of aggregation depend on this field but reading it via import is slow/expensive + additional issue of this being static code i.e some of the new modules after the server has started wont even exist